kits/research-paper-analyzer/apps/backend/main.py

```
@@ -0,0 +1,128 @@
+]
+
+
+class AnalyzeRequest(BaseModel):
+    pdf_url: str
+
+    @field_validator("pdf_url")
+    @classmethod
+    def validate_pdf_url(cls, v: str) -> str:
+        parsed = urlparse(v)
+        if parsed.scheme not in {"https"}:
+            raise ValueError("Only HTTPS URLs are accepted.")
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)